gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/SumGlobalStep.java

```diff
@@ -68,7 +69,48 @@ public void processAllStarts() {
 
     @Override
     public S projectTraverser(final Traverser.Admin<S> traverser) {
-        return (S) mul(traverser.get(), traverser.bulk());
+        final S value = traverser.get();
+        final long bulk = traverser.bulk();
+        final Number projected = mul(value, bulk);
+
+        // the following behavior allows global sum() to behave the same as sum(local)
+        if (value instanceof Long || value instanceof Double ||
```

Comment:
Float should probably be here too? I think that Long gets promoted to Float during addition, so Float is considered the "wider" type?